docs/0.4.0-alpha/basic_syntax/importing.md

```diff
@@ -59,6 +59,11 @@ main {
 }
 ```
 
+> DETAILS: Key features of `main` block:
+- Code outside the main block runs regardless of how the file is executed.
+- Code inside the main block runs only when the file is executed directly.
+- The `?` operator ensures that any exit code from some command is automatically passed back to the shell, making it easy to handle script results effectively.
```

Comment:
Do we state anywhere that this also applies to defined functions?

Reply:
Could you elaborate?